OpenVPN/openvpn-gui
# Pre-Logon Access Provider

At long last, here is  a Pre-Logon Access Provider for starting OpenVPN connections before logon.

Summary:
- First 3 commits make some minor tweaks to the GUI code (few lines)
- The major chunk of the code is COM interface implementations and some glue to re-use the GUI framework for enumerating and starting connections.

How to test:
- Copy libopenvpn_plap.dll to `C:\Program Files\OpenvPN\bin\`
- From an elevated prompt run `reg import openvpn-plap-install.reg`
- Have at least one profile in config-auto folder with the automatic service running 
- The config should have `management 127.0.0.1 <port> [pw-file]`   
  (`management-hold` is recommended, but not required)
- Have some option in the config that requires user input -- e.g.,`auth-user-pass` 
Lock screen, Logout or switch user to show the login screen, find the PLAP icon at bottom right, select it to see available connection profiles listed. Click on the profile tile and connect. 

Once logged in, the connection should show up in the GUI if using a GUI executable post PR #512.  

Some limitations: (i) management-query-proxy is limited to the use of system proxy setting, (ii) OPEN_URL is not supported.  (iii) there is no option to import a profile from the login screen.
